client.go

```diff
@@ -243,7 +246,7 @@ type Param struct {
 }
 
 func (q *Query) Add(new Param) {
-	ret := make(Query, len(*q))
+	ret := make(Query, 0, len(*q))
```

Comment:
My golang Achilles heel.

Reply:
seems like something golint could warn on, using `append` on a slice initialized with a non-zero length seems like it's always a bad idea.

Reply:
I just wish they had decided that make always was capacity first, then length.  Far too late now though.